packages/grafana-data/src/text/string.ts

```diff
@@ -1,6 +1,32 @@
+const specialChars = ['(', '[', '{', '}', ']', ')', '|', '*', '+', '-', '.', '?', '<', '>', '#', '&', '^', '$'];
+
+export const escapeStringForRegex = (value: string) => {
+  if (!value) {
+    return value;
+  }
+
+  return specialChars.reduce((escaped, currentChar) => escaped.replace(currentChar, '\\' + currentChar), value);
+};
+
+export const unEscapeStringFromRegex = (value: string) => {
+  if (!value) {
+    return value;
+  }
+
+  return specialChars.reduce((escaped, currentChar) => escaped.replace('\\' + currentChar, currentChar), value);
+};
+
+export function stringStartsAsRegEx(str: string): boolean {
+  if (!str) {
```

Comment:
What about `str === ''`?

Reply:
What about it :)

Reply:
I think this is more verbose than `!str` :) WDYT?

Reply:
My thinking is that this takes care of the case where one tries to call this with an undefined string. Should add a test for this perhaps?

Reply:
Hmm, TS should take care of this with the `str: string` typing

Reply:
Agree but this is string that comes from the UI, might not always be true that it is a string. Just being super cautious.

Reply:
Ok, that makes sense